### PR TITLE
Adds tests related to EV charging and parking selection for ride hail vehicles

### PR DESCRIPTION
**Pull request by HoneyTauOverTwo**
For reviewers: probably should review #3727 first, since this builds upon it.

Adds tests to Fully automated (Level 5) electric ride hail vehicles. Human driven ride hail vehicles are not covered because their charging point selection method is too primitive to be tested, human RH will, on most of the occasions, just charge on the closest charger (slack thread for details):

So to start the discussion about charging station selection for human driven ride hail electric vehicles, I guess the most important question is: How diverse are charging stations in respect to price and power rating on current simulations? If the answer is not diverse at all, and that if they are never planned to be diverse, there is not much need on updating the parking selection parameters.
Otherwise, there is a need to include more parameters on the selection for human driven EV RH. Currently, the selection parameters follows the same parameters used for private EV: (`ParkingMNL.Parameters.`)
`RangeAnxietyCost`, `WalkingEgressCost`, `ParkingTicketCost`, `HomeActivityPrefersResidentialParking`, `EnrouteDetourCost`
(relevant code: [ChargingFunctions.scala#L226](https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala#L226) and [ParkingFunctions.scala#L88](https://github.com/LBNL-UCB-STI/beam/blob/672960f66428b7cb843f7054c8f6026564d651c6/src/main/scala/beam/agentsim/infrastructure/ParkingFunctions.scala#L88))
The only relevant parameter for RH vehicles is `ParkingTicketCost`, all the others are 0 for human EV RH.
As opposed to automated EV RH: (`ParkingMNL.Parameters.`)
`DrivingTimeCost`, `QueueingTimeCost`, `ChargingTimeCost`, `InsufficientRangeCost`
(relevant code: [RideHailDepotFunctions.scala#L93](https://github.com/LBNL-UCB-STI/beam/blob/672960f66428b7cb843f7054c8f6026564d651c6/src/main/scala/beam/agentsim/infrastructure/RideHailDepotFunctions.scala#L93))
Which takes into consideration other relevant parameters that would make sense also being used by human EV RH.
Note that on the current implementation and using default parameter value for `beam.agentsim.agents.parking.minSearchRadius = 250` (in meters, this is 0.15 only miles), will only be choosing from the charging stations within this starting radius, unless there is none and the radius gets increased recursively.
In practice, this means that if there are only bad chargers (too expensive or slow charging) within the search radius, it will not consider picking chargers that might be just a bit further away but cheaper or faster to charge. I understand that this way of searching is done to improve performance, but if we were to add more parameters for the search, it only would make sense if the default value for minSearchRadius was increased.
Comparing to automated RH, the initial search radius for them is hardcoded as 40 Km (25 miles): [RideHailDepotNetwork.scala#L21](https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/scala/beam/agentsim/infrastructure/RideHailDepotNetwork.scala#L21). This makes sense since this will be searching only for privately owned chargers of the RH company so performance will not be affected as much, but still, the initial search radius for human EV RH is currently too small to allow other parameters to be relevant.

### CODE REVIEW

**Comment by HoneyTauOverTwo** [comment]
Commits from this PR start on the previous one from this comment. The other commits are related to PR #3727 for private vehicles.

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

File: src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package beam.agentsim.infrastructure
 
 import beam.agentsim.agents.vehicles.FuelType.FuelType
-import beam.agentsim.agents.vehicles.{BeamVehicle, BeamVehicleType, VehicleManager}
-import beam.agentsim.infrastructure.ParkingInquiry.ParkingActivityType.{Charge, Home, Work}
+import beam.agentsim.agents.vehicles.{BeamVehicleType, VehicleManager}
```
Comment (JustinPihony) [inline src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala]: Same comments as the other PR

Comment (HoneyTauOverTwo) [reply src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala]: I'll merge the changes from the other one.

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
test!

**Comment by HoneyTauOverTwo** [comment]
Flaky tests observed trying to get CI to pass:

`scripts.beam_to_matsim.io.WriterTest.testWriteViaEventsQueue`
`beam.utils.SnapCoordinateSpec.scenario plan should contain all valid locations`
`beam.agentsim.agents.RideHailAgentSpec.A RideHailAgent won't let me cancel its job after it has picked up passengers`

**Comment by HoneyTauOverTwo** [comment]
test!

File: src/test/scala/beam/agentsim/agents/ElectricVehicleChargingBehaviorTest.scala [hunk 2]
```diff
@@ -238,7 +292,230 @@ class ElectricVehicleChargingBehaviorTest
...
+      count + pluginCountByTAZ.getOrElse(taz, 0)
+    }
+    val farTazPluginCAVCount = furtherTAZsCAV.foldLeft(0) { (count, taz) =>
+      count + pluginCountByTAZ.getOrElse(taz, 0)
+    }
+
+    closeTazPluginCAVCount should be > farTazPluginCAVCount withClue
+    ", vehicles should be picking the closest charger more often than the farther ones."
+
+    val closeTazPluginHumanCount = closestTAZsHuman.foldLeft(0) { (count, taz) =>
```
Comment (JustinPihony) [inline src/test/scala/beam/agentsim/agents/ElectricVehicleChargingBehaviorTest.scala]: These should be commented out/removed too - since they are not used

**Review by JustinPihony — APPROVED** [review]

**Comment by HoneyTauOverTwo** [comment]
test!

**Pull request merged**